File: string/GroupAna/groupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    var map = {}
    var result = []
    for (var i=0; i<strs.length; i++){
        var key = strs[i].split('').sort();
        if(map[key]===undefined){
            map[key] = []
            map[key].push(strs[i])
        }else{
            map[key].push(strs[i])
        }
    }
    for(var i in map){
        result.push(map[i]);
    }
    return result
};

File: string/GroupAna/groupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if(strs==null) return null;
        HashMap<String,List<String>> map = new HashMap<>();
        for (int i=0;i<strs.length;i++){
            char [] key = strs[i].toCharArray();
            Arrays.sort(key);
            String KeyValue = String.valueOf(key);
            if(!map.containsKey(KeyValue)){
                map.put(KeyValue,new ArrayList<>());

            }
                map.get(KeyValue).add(strs[i]);
        }
        return new ArrayList<>(map.values());
     }
}
